modules/core/src/scenegraph/gltf-instantiator/gltf-animator.js

```diff
@@ -0,0 +1,95 @@
+import {Matrix4} from 'math.gl';
+
+// TODO: import from loaders.gl?
+export const ATTRIBUTE_TYPE_TO_COMPONENTS = {
+  SCALAR: 1,
+  VEC2: 2,
+  VEC3: 3,
+  VEC4: 4,
+  MAT2: 4,
+  MAT3: 9,
+  MAT4: 16
+};
+
+export const ATTRIBUTE_COMPONENT_TYPE_TO_ARRAY = {
+  5120: Int8Array,
+  5121: Uint8Array,
+  5122: Int16Array,
+  5123: Uint16Array,
+  5125: Uint32Array,
+  5126: Float32Array
+};
+//
+
+
+// TODO: save and do not regenerate
+function accessorToJsArray(accessor) {
+  const ArrayType = ATTRIBUTE_COMPONENT_TYPE_TO_ARRAY[accessor.componentType];
+  const components = ATTRIBUTE_TYPE_TO_COMPONENTS[accessor.type];
+  const length = components * accessor.count;
+  const {buffer, byteOffset} = accessor.bufferView.data;
+
+  return {array: new ArrayType(
+    buffer,
+    byteOffset + (accessor.byteOffset || 0),
+    length
+  ), components};
+}
+
+// TODO: share with GLTFInstantiator
+function applyTRS(gltfNode, node) {
+  node.matrix.identity();
+
+  if (gltfNode.translation) {
+    node.matrix.translate(gltfNode.translation);
+  }
+
+  if (gltfNode.rotation) {
+    const rotationMatrix = new Matrix4().fromQuaternion(gltfNode.rotation);
```

Comment:
No need to allocate a Matrix4 here. Since it's always being overwritten, you can declare it outside the function and re-use it.